File: src/types/financialHealth.ts
// ── Financial Health types ───────────────────────────────────────────────────
export interface HealthMetric {
  label: string;
  score: number;   // 0–100
  weight: number;  // contribution to overall
  description: string;
}

export interface FinancialHealth {
  overallScore: number;   // 0–100
  metrics: {
    savingsRate: HealthMetric;
    budgetAdherence: HealthMetric;
    spendingStability: HealthMetric;
    debtBurden: HealthMetric;
    goalProgress: HealthMetric;
    emergencyReserve: HealthMetric;
  };
  previousScore: number;
  scoreChange: number;
  aiExplanation: string;
  generatedAt: string;
}

File: backend/app/schemas/financial_health.py
from pydantic import BaseModel
from typing import Dict


class HealthMetricOut(BaseModel):
    label: str
    score: float
    weight: float
    description: str


class FinancialHealthOut(BaseModel):
    overallScore: float
    previousScore: float
    scoreChange: float
    metrics: Dict[str, HealthMetricOut]
    aiExplanation: str
    generatedAt: str
